slashing-protection/src/integration-test/java/tech/pegasys/web3signer/slashingprotection/PruningRunnerIntegrationTest.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import dsl.TestSlashingProtectionParameters;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PruningRunnerIntegrationTest extends IntegrationTestBase {
+
+  public static final String PRUNING_THREAD_NAME = "slashing-db-pruner-0";
+  private ScheduledExecutorService scheduledExecutorService;
+  private TestSlashingProtectionParameters slashingProtectionParameters;
+  private SlashingProtection pruningSlashingProtection;
+
+  @BeforeEach
+  void setupSlashingProtection() {
+    slashingProtectionParameters =
+        new TestSlashingProtectionParameters(databaseUrl, USERNAME, PASSWORD, 5, 1);
+    scheduledExecutorService =
```

Comment:
do we need an `@AfterEach` to stop/close the executor?

Reply:
yeah should do. done.